### ISSUE TITLE
@Dataset transactional: How does it works?

### ISSUE THREAD

**Issue by duschata** [database-rider/database-rider#356]
Hello All, hello Rafael,

please have a look at these lines of code:

https://github.com/duschata/testingIssues/blob/417cb9b9ea2465c2b73ee879b63f18f805ce4c69/src/junit5Test/java/riderIssue/BRiderJunit5InANutshellTest.java#L41-L61

The behavior is not what I expect. With the attribute transactional=true, I expect, that a transaction is open within the testfunction.

Kind regards,
Tom

**Comment by rmpestano** [comment]
Hi, the commit is done after the test, check docs here: https://github.com/database-rider/database-rider#315-transactional-tests

**Comment by duschata** [comment]
Sorry, but Idon't understand your answer.  There is no active transaction passing the testfunction. A commit after the test would fail. The @Test is JUnit5 maybe this works only with junit4.

**Comment by duschata** [comment]
no doubt about it. It works in this test. There is the deprecated hibernate-entitymanager  used. We are using the EM from hibernate-core which is the recommended way. If you delete it from your pom.xml (rider-parent) the test will use the EM from core and will also fail.

**Comment by rmpestano** [comment]
I'll try to have a look, thanks for the feedback

**Comment by rmpestano** [comment]
Check my PR on your sample project: https://github.com/duschata/testingIssues/pull/1

I hope it makes more sense

**Comment by duschata** [comment]
Sorry for my confusing code, I made it a little bit clearer now. The essential statement of these lines where: ups, there is no transaction active. It tooks some time until I understand why your fix works. There are two tastings of the EntityManagerProvider. One in the junit5 package and the other in core. I've used the wrong one. And this has realy nothing to do with the deprecated hibernate-entitymanager.  But maybe this must be a little clearer in the documentation docs. So far so good. I'm on weekend now and I wish you also a nice one...
